src/libp2p_transport_tcp.erl

```diff
@@ -85,13 +104,18 @@ tcp_listen_addrs(Socket) ->
         true ->
             % all 0 address, collect all non loopback interface addresses
             {ok, IFAddrs} = inet:getifaddrs(),
-            [multiaddr({Addr, Port}) ||
+            ListenAddrs = [multiaddr({Addr, Port}) ||
                 {_, Opts} <- IFAddrs, {addr, Addr} <- Opts, {flags, Flags} <- Opts,
                 size(Addr) == size(IP),
                 not lists:member(loopback, Flags),
                 %% filter out ipv6 link-local addresses
                 not (size(Addr) == 8 andalso element(1, Addr) == 16#fe80)
-            ]
+            ],
+            Parent = self(),
```

Comment:
I certainly wasn't thinking of kicking these of here? Perhaps move them to where start_listener happens?